gunrock/app/bc/bc_enactor.cuh

```diff
@@ -491,44 +487,42 @@ template <
 class Enactor :
     public EnactorBase<
         typename _Problem::GraphT,
-        typename _Problem::GraphT::VertexT, // TODO: change to other label types used for the operators, e.g.: typename _Problem::LabelT,
-        typename _Problem::GraphT::ValueT, // TODO: change to other value types used for inter GPU communication, e.g.: typename _Problem::ValueT,
+        typename _Problem::GraphT::VertexT,
+        typename _Problem::ValueT,
         ARRAY_FLAG, cudaHostRegisterFlag>
 {
 public:
     typedef _Problem                   Problem ;
     typedef typename Problem::SizeT    SizeT   ;
     typedef typename Problem::VertexT  VertexT ;
     typedef typename Problem::GraphT   GraphT  ;
-    // TODO: change according to the EnactorBase template parameters above
     typedef typename GraphT::VertexT   LabelT  ; // e.g. typedef typename Problem::LabelT LabelT;
-    typedef typename GraphT::ValueT    ValueT  ; // e.g. typedef typename Problem::ValueT ValueT;
-    typedef EnactorBase<GraphT , LabelT, ValueT, ARRAY_FLAG, cudaHostRegisterFlag>
+    typedef typename Problem::ValueT    ValueT  ; // e.g. typedef typename Problem::ValueT ValueT;
+    typedef EnactorBase<GraphT , VertexT, ValueT, ARRAY_FLAG, cudaHostRegisterFlag>
         BaseEnactor;
     typedef Enactor<Problem, ARRAY_FLAG, cudaHostRegisterFlag>
         EnactorT;
-    
-    typedef BCForwardIterationLoop<EnactorT> ForwardIterationT;
+
+    typedef BCForwardIterationLoop <EnactorT> ForwardIterationT;
     typedef BCBackwardIterationLoop<EnactorT> BackwardIterationT;
 
-    Problem     *problem   ;
-    ForwardIterationT  *forward_iterations;
+    Problem             *problem   ;
+    ForwardIterationT   *forward_iterations;
     BackwardIterationT  *backward_iterations;
 
     /**
-     * @brief SSSPEnactor constructor
+     * @brief BCEnactor constructor
      */
     Enactor() :
         BaseEnactor("bc"),
         problem    (NULL  )
     {
-        // TODO: change according to algorithmic needs
         this -> max_num_vertex_associates = 0;
-        this -> max_num_value__associates = 1;
+        this -> max_num_value__associates = 2;
```

Comment:
What does this do?

Reply:
This is telling the multi-GPU framework to allocate 2 associative values of type ValueT, and 0 associative values of type VertexT for each vertex in the remote sub-frontier for communication. These two values are related to the template parameters of `Iteration_Loop`

Reply:
What is an associative value exactly?